### DISCUSSION TITLE
How do I convert TipTap JSON to markdown?

### DISCUSSION THREAD

**Discussion by punkpeye** [ueberdosis/tiptap#5847] [General Questions] [answered]
I am using TipTap in https://glama.ai/.

I need to send messages to AI using markdown.

At the moment, I am doing TipTap -> HTML -> Markdown.

However, I am finding various edge cases where this does not work well, e.g. tables.

Is there a better way of doing this?

**Comment by nperez0111** [comment, marked as answer]
HTML -> markdown is probably the best bet right now. You can also extend extensions to customize the html output just for the html export if you need to. I do have a PR for a static-renderer which could do it but I've not finished it yet: https://github.com/ueberdosis/tiptap/pull/5528

**Reply by punkpeye** [reply]
@romansp this would be a major improvement to the existing flow. I appreciate you working on it! Any timeline you could share, like days, weeks, months?

**Reply by nperez0111** [reply]
I should also mention that you can use a markdown serializer for this too, here is a quick one I threw together: https://github.com/ueberdosis/tiptap/compare/develop...markdown

**Reply by punkpeye** [reply]
oh hell yeah, I will give this a try

**Reply by punkpeye** [reply]
@nperez0111 What's the `inAutolink` that you have in that snippet?

Everything else make sense, but that's the one that's throwing a type error and I am not sure what it is for.

**Reply by nperez0111** [reply]
Most of it was pulled from here: https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/to_markdown.ts

**Reply by punkpeye** [reply]
I ended up just copying most of that and it works pretty well!! Thank you
